freqtrade/misc.py

```diff
@@ -114,3 +115,23 @@ def deep_merge_dicts(source, destination):
             destination[key] = value
 
     return destination
+
+
+def find_project_root(path=None):
```

Comment:
I'm not sure if this will work correctly (honestly, i think it'll fail for most cases)

Your assumption is that you've got the repository cloned and you're in a jupyter notebook within `user_data/`.

As pointed out in #2112 - one goal is for `freqtrade create-datadir` to copy the contents of the current user_dir to the newly created directory, so users have the examples available.
While it's not doing that yet, it's not a huge step to implement that (and it's on my todo-list ... among other things)
Also, #2112 's goal is to install from pypi - in which case a `git clone` won't even happen

Assuming i install freqtrade reguilarily then cd to ~ and create a userdir there:

``` bash
pip install .[all]
# pip install freqtrade  ## for the pypi future 
cd ~
freqtrade create-userdir --userdir ~/ft_userdata/
```

I can now run the bot with `--userdir ~/ft_userdata/` - and i can version-control this folder in my private repo (it contains all my profitable strategies i'm not ready to share...)

I also have the notebook samples there - but i cannot find the projects-directory - since its 

a) not relevant - it's somewhere in `/usr/.../pip/site-packages/...`
b) not one of my parents

Reply:
I'll pull this out. I can see how fragile this can be.

Reply:
btw, i think you somehow messed the 2 branches for this and #2222  - they seem to have the same (similar?) changes now ...

Reply:
It is very possible that I mangled something. Do you want me to start a new PR with the useful changes on develop or, can you accept this and I'll rebase the other on top of the new develop?

Reply:
Accepting this in it's current state won't be possible since it has import-errors which would prevent the bot from running.

i think it's best to start a new branch from develop and then forcepush this on top of this so we don't loose comments and such.

to not loose your changes best do the following:

``` bash
git checkout -b new_notebook  # create new branch
git merge notebook --no-commit # add changes from this branch - without commiting
git reset  # reset state to "dirty" - most tools (vscode for example) will now highlight changed lines

### Now remove all changes that are unwanted
### then:

git add ... # whatever files are still left
git commit

# you can then rename the branches and force-push on top of this
git branch -m notebook notebook_old
git branch -m notebook_new notebook
git push -u origin notebook # Make sure to replace origin with the remote of your git repository

```

Reply:
Thanks for the coaching. I think the last push fixed it. Hopefully you can accept these changes.